front/src/routes/scene/edit-scene/actions/CheckTime.jsx

```diff
@@ -0,0 +1,134 @@
+import Select from 'react-select';
+import { Component } from 'preact';
+import DatePicker from 'react-datepicker';
+import { connect } from 'unistore/preact';
+import { format } from 'date-fns';
+import { Text, Localizer } from 'preact-i18n';
+
+@connect('', {})
+class CheckTime extends Component {
+  handleBeforeTimeChange = time => {
+    const timeFormatted = time ? format(time, 'HH:mm') : undefined;
+    this.props.updateActionProperty(this.props.columnIndex, this.props.index, 'before', timeFormatted);
+  };
+  handleBeforeAfterChange = time => {
+    const timeFormatted = time ? format(time, 'HH:mm') : undefined;
+    this.props.updateActionProperty(this.props.columnIndex, this.props.index, 'after', timeFormatted);
+  };
+  handleDayOfTheWeekChange = options => {
+    const values = options ? options.map(option => option.value) : undefined;
+    this.props.updateActionProperty(this.props.columnIndex, this.props.index, 'days_of_the_week', values);
+  };
+
+  render(props, state) {
+    const { localeSet } = state;
+    const before = this.props.action.before
+      ? new Date().setHours(this.props.action.before.substr(0, 2), this.props.action.before.substr(3, 2))
+      : null;
+    const after = this.props.action.after
+      ? new Date().setHours(this.props.action.after.substr(0, 2), this.props.action.after.substr(3, 2))
+      : null;
+    const daysOfTheWeekOptions = [
+      {
+        value: 'monday',
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.monday" />
+      },
+      {
+        value: 'tuesday',
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.tuesday" />
+      },
+      {
+        value: 'wednesday',
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.wednesday" />
+      },
+      {
+        value: 'thursday',
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.thursday" />
+      },
+      {
+        value: 'friday',
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.friday" />
+      },
+      {
+        value: 'saturday',
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.saturday" />
+      },
+      {
+        value: 'sunday',
+        label: <Text id="editScene.triggersCard.scheduledTrigger.daysOfTheWeek.sunday" />
+      }
```

Comment:
In `ScheduledTrigger.jsx` you
```
 const selectedWeekDaysOptions = this.props.trigger.days_of_the_week
      ? this.props.trigger.days_of_the_week.map(day => ({
          value: day,
          label: <Text id={`editScene.triggersCard.scheduledTrigger.daysOfTheWeek.${day}`} />
        }))
      : [];
```

Any reason to list it explicitly here?

Reply:
No, in scheduled trigger it's listed explicity as well: 

<img width="802" alt="Screenshot 2021-04-22 at 09 51 51" src="https://user-images.githubusercontent.com/7365207/115644013-6577e700-a350-11eb-80f6-983f720de9f9.png">

But you're right we could make a small array and .map, it'll be less verbose :)

Reply:
Fixed